### PR TITLE
Support passing simulator udid

### PR DESCRIPTION
**Pull request by keith**
Previously we supported passing the name of a specific simulator. This
adds support for passing a udid directly as well.

### CODE REVIEW

File: sources/cli.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+func consumeArgument(flag: String, from arguments: inout ArraySlice<String>) -> String? {
```
Comment (sberrevoets) [inline sources/cli.swift]: Your call, but `flag` implies `Bool` in my mind. `name` might be more descriptive?

File: sources/cli.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    if let argumentIndex = arguments.index(of: flag) {
+        let value = arguments.suffix(from: argumentIndex).dropFirst().joined(separator: " ")
+        if value.isEmpty {
+            exitWithUsage()
+        }
+
+        arguments = arguments.prefix(upTo: argumentIndex)
+        return value
+    }
```
Comment (sberrevoets) [inline sources/cli.swift]: Would `guard let` here to reduce indentation

File: sources/errors.swift [hunk 3]
```diff
@@ -1,8 +1,22 @@
...
+        }
+    }
+}
+
 enum SimulatorFetchError: Error, CustomStringConvertible {
     case simctlFailed
     case failedToReadOutput
     case noBootedSimulators
     case noMatchingSimulators(name: String)
+    case noMatchingUDID(udid: UUID)
```
Comment (sberrevoets) [inline sources/errors.swift]: Not sure if intentional, but might want to double check UDID vs. UUID usage

Comment (keith) [reply sources/errors.swift]: It is intentional, but it's still confusing. I would use `udid` everywhere but the `UUID` type makes that difficult. But from the simctl json it's `udid`

File: sources/cli.swift (hunk 1 above)
Comment (keith) [reply sources/cli.swift]: done

File: sources/cli.swift (hunk 2 above)
Comment (keith) [reply sources/cli.swift]: done

**Pull request merged**